packages/region/src/services/region-module.ts

```diff
@@ -223,37 +183,167 @@ export default class RegionModuleService<
-      result = await this.regionService_.update(
-        [{ id: idOrSelectorOrData, ...data }],
-        sharedContext
-      )
+      normalizedInput = [{ id: idOrSelectorOrData, ...data }]
     }
 
     if (Array.isArray(idOrSelectorOrData)) {
-      result = await this.regionService_.update(
-        idOrSelectorOrData,
-        sharedContext
-      )
+      normalizedInput = idOrSelectorOrData
     }
 
     if (isObject(idOrSelectorOrData)) {
-      let toUpdate: Partial<UpdateRegionDTO>[] = []
       const regions = await this.regionService_.list(
-        { ...idOrSelectorOrData },
+        idOrSelectorOrData,
         {},
         sharedContext
       )
 
-      regions.forEach((region) => {
-        toUpdate.push({ id: region.id, ...data })
+      normalizedInput = regions.map((region) => ({
+        id: region.id,
+        ...data,
+      }))
+    }
+    normalizedInput = this.normalizeInput(normalizedInput)
+
+    // If countries are being updated for a region, first make previously set countries' region to null to get to a clean slate.
+    // Somewhat less efficient, but region operations will be very rare, so it is better to go with a simple solution
+    await this.countryService_.update(
+      {
+        selector: {
+          region_id: normalizedInput
+            .filter((region) => !!region.countries)
+            .map((region) => region.id)
+            .flat(),
+        },
+        data: { region_id: null },
+      },
+      sharedContext
+    )
+
+    const validations = [
+      this.validateCurrencies(
+        normalizedInput.map((d) => d.currency_code),
+        sharedContext
+      ),
+      this.validateCountries(
+        normalizedInput.map((d) => d.countries ?? []).flat(),
+        sharedContext
+      ),
+    ] as const
+
+    // Assign the full country object so the ORM updates the relationship
+    const [, dbCountries] = await Promise.all(validations)
+    const dbCountriesMap = asMap(dbCountries, "iso_2")
+    let normalizedDbRegions = normalizedInput.map((region) =>
+      removeUndefined({
+        ...region,
+        countries: region.countries?.map((c) => dbCountriesMap[c]),
+      })
+    )
+
+    return await this.regionService_.update(normalizedDbRegions, sharedContext)
+  }
+
+  private normalizeInput<T extends UpdatableRegionFields>(regions: T[]): T[] {
+    return regions.map((region) =>
+      removeUndefined({
+        ...region,
+        currency_code: region.currency_code?.toLowerCase(),
+        name: region.name?.trim(),
+        countries: region.countries?.map((country) => country.toLowerCase()),
       })
+    )
+  }
 
-      result = await this.regionService_.update(toUpdate, sharedContext)
+  @InjectTransactionManager("baseRepository_")
+  private async validateCurrencies(
+    currencyCodes: (string | undefined)[] | undefined,
+    sharedContext: Context
+  ): Promise<void> {
+    const normalizedCurrencyCodes = currencyCodes
+      ?.filter((c) => c !== undefined)
```

Comment:
todo: here null will go to the map step. 
```ts
currencyCodes
      ?.filter((c) => !!c)
      .map(c => c.toLowerCase()
```

Reply:
Well actually I was trying to handle the scenario where someone passes an empty string, for example. If they do, the validation below will catch it, but if we filter it out, it won't, so checking undefined only is better.

Reply:
Yes exactly @olivermrbl